recipes/lnav/variants.yaml

```diff
@@ -0,0 +1,4 @@
+MACOSX_DEPLOYMENT_TARGET:  # [osx and x86_64]
+  - 10.15                  # [osx and x86_64]
```

Comment:
```suggestion
```

c_stdlib_version should be sufficient. Can you also try conda_build_config.yaml? I don't know whether variants gets correctly picked up. cc @pavelzw @wolfv

Reply:
I also tried conda_build_config.yaml but the build failed.

Reply:
In this case, is it correct to use conda_build_config.yaml or variants.yaml?

I know I can use both in ratter-build, but what is the real difference between these two files?

Reply:
I changed build system from rattler-build to conda-build with conda_build_config.yaml. Now building passes.

Reply:
@xhochy 

Ready to be merged.